### ISSUE TITLE
Static fields for rules are not correctly parsed

### ISSUE THREAD

**Issue by albertomn86** [wazuh/wazuh#3407]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.9.1 | ossec-analysisd | manager | Packages/Sources | Ubuntu 17.10 |

The fields `protocol` and `system_name` are not being recognized in the rules parsing.
These fields are blocked to be used as dynamic fields, but they are not being read from the rules XML files.

https://github.com/wazuh/wazuh/blob/090986e4bebbfc737c551d6b4d9485acfb9d1a5c/src/analysisd/rules.c#L623

https://github.com/wazuh/wazuh/blob/090986e4bebbfc737c551d6b4d9485acfb9d1a5c/src/analysisd/rules.c#L630

This issue was reported in the Wazuh mailing list: https://groups.google.com/forum/#!topic/wazuh/hu7-zXqZgoA

**Comment by carlocas** [comment]
**Issue update**
The fields `protocol` and `system_name` are being recognized in the rules. But i discover that the fields `data` and `extra_data` are not being recognized, so the issue remain in progress. I am working in the branch `issue-3407`.

**Comment by crolopez** [comment]
The current state of the solution implemented in (https://github.com/wazuh/wazuh/issues/3407) solves the problem by default, taking into account only the events whose agent ID matches. However, there is an option that evaluates [the location field](https://documentation.wazuh.com/3.9/user-manual/ruleset/ruleset-xml-syntax/rules.html?highlight=rule%20syntax#same-location), which is a key formed by the name of the agent, its ID and the origin of the event. 

We have to decide on the following:
- If we think that the `<same_location>` tag is enough, we will have to add this option to all the conceptual rules (223 currently). We will also need to extend its documentation.
- If we think that the `<same_location>` tag is not enough, either because users don't usually associate the location with the agent identity and we need an explicit way to name it, or because including the origin of the event in this primary key can generate invalid behaviours:
  - We can add the agent ID check by default and allow the user to block this check by another option like `<global_frequency>`, where the word global means that the engine will take into account the global counter of all agents to evaluate the alert triggering.
  - We can allow the check only if an explicit option specifies it, such as `<same_agent_id>`. This option would make us modify all the contextual rules.
  - We can also add an attribute to the `frequency` tag level that specifies it. This option would also make us modify all the implicated rules.
``` XML
  <rule id="11306" level="10" frequency="8" timeframe="3600" global="yes">
```

**Comment by chemamartinez** [comment]
Merged #3591
